radixdlt/src/integration/java/com/radixdlt/consensus/deterministic/ControlledNode.java

```diff
@@ -67,34 +68,45 @@
 	private final ValidatorSet initialValidatorSet;
 	private final ControlledSender controlledSender;
 
+	public enum SyncAndTimeout {
+		NONE,
+		SYNC,
+		SYNC_AND_TIMEOUT
+	}
+
 	ControlledNode(
 		String name,
 		ECKeyPair key,
 		ControlledSender sender,
 		ProposerElectionFactory proposerElectionFactory,
 		ValidatorSet initialValidatorSet,
-		boolean enableGetVerticesRPC,
+		SyncAndTimeout syncAndTimeout,
 		SyncedStateComputer<CommittedAtom> stateComputer
 	) {
 		this.systemCounters = new SystemCountersImpl();
 		this.controlledSender = Objects.requireNonNull(sender);
 		this.initialValidatorSet = Objects.requireNonNull(initialValidatorSet);
 
-
-		SyncVerticesRPCSender syncVerticesRPCSender = enableGetVerticesRPC ? sender : UnsupportedSyncVerticesRPCSender.INSTANCE;
+		SyncVerticesRPCSender syncVerticesRPCSender = (syncAndTimeout != SyncAndTimeout.NONE)
+			? sender
+			: UnsupportedSyncVerticesRPCSender.INSTANCE;
+		LocalTimeoutSender localTimeoutSender = (syncAndTimeout == SyncAndTimeout.SYNC_AND_TIMEOUT) ? sender : (v, t) -> { };
 		Mempool mempool = new EmptyMempool();
-		Hasher hasher = new DefaultHasher();
+		Hasher nullHasher = data -> Hash.ZERO_HASH;
```

Comment:
hmm `nullHasher` is not exaclty correct though? Since we don't return `null`, but rather `ZERO_HASH`, so maybe rename it `dummyHasher`? or `alwaysZeroHasher`?

Reply:
Not part of PR.

Reply:
Think "null" in the sense that it doesn't do any hashing.